packages/astro/src/runtime/server/render/component.ts

```diff
@@ -93,6 +93,7 @@ async function renderFrameworkComponent(
 	};
 
 	const { hydration, isPage, props } = extractDirectives(_props, clientDirectives);
+	const propsWithoutTransitionAttributes = withoutTransitionAttributes(props);
```

Comment:
Could we move the transition attributes filtering inside `extractDirectives` instead? That makes it a little and tucks in the transition attribute code in one file.

Reply:
Yeah I think the `extractDirectives` naming is fine. If we want to change it, something like `extractPropStates` would also be fine by me